File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
    vector<int> v;
    int n = numbers.size()-1;
    for(int i=0 ; i<numbers.size(); i++){
        int key = target - numbers[i];
        int start=i+1;
        int end = n;
        int idx = 1;
        while(start<=end){
            int mid = (start+end)/2;
            if(numbers[mid]==key){
                idx = mid;
                break;
            }
            else if(numbers[mid]<key){
                start=mid+1;
            }
            else if(numbers[mid]>key){
                end=mid-1;
            }
        }
        if(numbers[i]+numbers[idx]==target){
            v.push_back(i+1);
            v.push_back(idx+1);
            break;
        }
    }
        return v;
    }
};

File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.java
class Solution {
    
    public static int BinarySearch(int[] arr  , int start , int end , int key){
        while(start<=end){
            int mid = start + (end-start)/2;
            if(arr[mid]==key){
                return mid;
            }
            else if(arr[mid]<key){
                start=mid+1;
            }
            else{
                end = mid-1;
            }
        }
        return -1;
    }
    
    public int[] twoSum(int[] numbers, int target) {
        int[] ans = new int[2];
        for(int i=0 ; i<numbers.length ; i++){
            int key = target - numbers[i];
            int idx = BinarySearch(numbers , i+1 , numbers.length-1 , key);
            if(idx>-1){
                ans[0] = i+1;
                ans[1] = idx+1;
                break;
            }
        }
        return ans;
    }
}